src/services/session.service.ts

```diff
@@ -1,64 +1,49 @@
 import { ErrorsMessages } from '@constants/errorMessages';
 import { Service } from 'typedi';
-import { getRepository } from 'typeorm';
 import { User } from '@entities/user.entity';
 import { UsersService } from '@services/users.service';
 import { RedisService } from '@services/redis.service';
+import { JWTService } from '@services/jwt.service';
 import { AuthInterface } from '@interfaces';
 import { DatabaseError } from '@exception/database.error';
 import { RedisError } from '@exception/redis.error';
 import { HttpError } from 'routing-controllers';
 import { HttpStatusCode } from '@constants/httpStatusCode';
+import { BaseUserDTO } from '@dto/baseUserDTO';
 
 @Service()
 export class SessionService {
   constructor(
     private readonly userService: UsersService,
-    private readonly redisService: RedisService
+    private readonly redisService: RedisService,
+    private readonly jwtService: JWTService
   ) {}
 
-  private readonly userRepository = getRepository<User>(User);
-
   async signUp(user: User) {
-    this.userService.hashUserPassword(user);
     try {
-      return await this.userRepository.save(user);
+      return await this.userService.createUser(user);
```

Comment:
Please do not use return await.

When you declare an async function, it automatically envelops the return of that function as a promise if it is not a promise already. 

Example: 

```javascript
async getName(): Promise<string> {
   return 'ernesto';
}
```
The example shown above does not return a string. It returns a promise that resolves in a string value. 


On the other hand, when you use await inside in an async function you are expecting that function to be executed from the micro-task queue on the Event Loop, and that's results in resolving the promised value. 

Example:

```javascript
async printer(): Promise<string> {
   const name: string = await getName(); -------> /// What you are doing here is expecting the promise to resolve so you have a plain string value and not a Promise that resolves in a string (Promise<string>)
   return name;
}
```

So what happens when you use "return await" inside an async function is basically adding additional delay because await pauses the execution until the promise is resolved (or rejected) and then, as you are doing this inside an async function you are forcing to re-promise the value (basically you have a promise, waits it to resolve, and then you are returning a promise again because that's the nature of async functions)

Reply:
fixed